server/config/config.go

```diff
@@ -107,10 +109,33 @@ func (conf *Config) Reconfigure(stored StoredConfig, mmconf *model.Config, licen
 		return err
 	}
 
+	var localURL string
+	if mmconf.ServiceSettings.ConnectionSecurity != nil && *mmconf.ServiceSettings.ConnectionSecurity == model.ConnSecurityTLS {
+		// If there is no reverse proxy use the server URL
+		localURL = *mattermostSiteURL
+	} else {
+		// Avoid the reverse proxy by using the local port
+		listenAddress := mmconf.ServiceSettings.ListenAddress
+		if listenAddress == nil {
+			return errors.New("plugin requires Mattermost Listen Address to be set")
+		}
+		host, port, err := net.SplitHostPort(*listenAddress)
+		if err != nil {
+			return err
+		}
+
+		if host == "" {
+			host = "127.0.0.1"
+		}
+
+		localURL = "http://" + host + ":" + port
```

Comment:
Is it possible that `port` is zero?

Reply:
Yes, that is possible, but I would count that as a bug of the server: https://github.com/mattermost/mattermost-server/blob/92837fa1ee69c4420ea0033a35b5058ca8afc76f/model/config.go#L3692-L3702